src/base/types/wrappers/SMomentModel.jl

```diff
@@ -1,132 +1,159 @@
+n_reactions(model::SMomentModel) = length(model.columns)
 
 """
-    n_reactions(model::SMomentModel)
+    bounds(model::SMomentModel)
 
-Returns the number of irreversible reactions in `model`.
+Return the variable bounds for [`SMomentModel`](@ref).
 """
-n_reactions(model::SMomentModel) = length(model.irrev_reaction_ids)
+bounds(model::SMomentModel) =
+    ([col.lb for col in model.columns], [col.ub for col in model.columns])
 
 """
-    metabolites(model::SMomentModel)
+    reaction_flux(model::SMomentModel)
 
-Return the metabolites in `model`.
+Get the mapping of the reaction rates in [`SMomentModel`](@ref) to the original
+fluxes in the wrapped model.
 """
-metabolites(model::SMomentModel) = model.metabolites
+reaction_flux(model::SMomentModel) =
+    _smoment_column_reactions(model)' * reaction_flux(model.inner)
 
 """
-    n_metabolites(model::SMomentModel) = 
+    coupling(model::SMomentModel)
 
-Return the number of metabolites in `model`.
+Return the coupling of [`SMomentModel`](@ref). That combines the coupling of
+the wrapped model, coupling for split reactions, and the coupling for the total
+enzyme capacity.
 """
-n_metabolites(model::SMomentModel) = length(metabolites(model))
+coupling(model::SMomentModel) = vcat(
+    coupling(model.inner) * _smoment_column_reactions(model),
+    _smoment_reaction_coupling(model),
+    [col.capacity_required for col in model.columns]',
+)
 
 """
-    bounds(model::SMomentModel)
+    n_coupling_constraints(model::SMomentModel)
 
-Return variable bounds for `SMomentModel`.
+Count the coupling constraints in [`SMomentModel`](@ref) (refer to
+[`coupling`](@ref) for details).
 """
-bounds(model::SMomentModel) = (model.xl, model.xu)
+n_coupling_constraints(model::SMomentModel) =
+    n_coupling_constraints(model.inner) + length(model.coupling_row_reaction) + 1
 
 """
-    reaction_flux(model::MetabolicModel)
+    coupling_bounds(model::SMomentModel)
 
-Helper function to get fluxes from optimization problem.
+The coupling bounds for [`SMomentModel`](@ref) (refer to [`coupling`](@ref) for
+details).
 """
-function reaction_flux(model::SMomentModel)
-    R = spzeros(n_fluxes(model), n_reactions(model) + 1)
-    for (i, rid) in enumerate(fluxes(model))
-        for_idx = findfirst(
-            x -> x == rid * "§ARM§FOR" || x == rid * "§FOR",
-            model.irrev_reaction_ids,
-        )
-        rev_idx = findfirst(
-            x -> x == rid * "§ARM§REV" || x == rid * "§REV",
-            model.irrev_reaction_ids,
-        )
-        !isnothing(for_idx) && (R[i, for_idx] = 1.0)
-        !isnothing(rev_idx) && (R[i, rev_idx] = -1.0)
-    end
-    return R'
+function coupling_bounds(model::SMomentModel)
+    (iclb, icub) = coupling_bounds(model.inner)
+    (ilb, iub) = bounds(model.inner)
+    return (
+        vcat(iclb, ilb[model.coupling_row_reaction], [0.0]),
+        vcat(icub, iub[model.coupling_row_reaction], [model.total_enzyme_capacity]),
+    )
```

Comment:
Also don't get this

Reply:
it follows the structure of the coupling,
- iclb/icub are bounds for the original model's coupling
- ilb/iub are bounds for the "not arm" reactions, taken from the original model
- 0.0 and total_enzyme_capacity are bounds for total enzyme capacity aka the sMoment row

Reply:
this makes sense now :)